### PR TITLE
DataSources: Introduce status field to api/ds/query for HTTP 207 Multi Status responses

### PR DESCRIPTION
**Pull request by wbrowne**
**What this PR does / why we need it**:
Depends on https://github.com/grafana/grafana-plugin-sdk-go/pull/510

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/api/metrics_response.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	jsoniter "github.com/json-iterator/go"
+
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+)
+
+type metricsResponse struct {
+	Results results
+}
+
+type results map[string]queryResponse
```
Comment (ryantxu) [inline pkg/api/metrics_response.go:15]: why not just use [QueryDataResponse](https://github.com/grafana/grafana-plugin-sdk-go/blob/main/backend/data.go#L75)

or is this just part of the draft?

File: pkg/api/metrics.go [hunk 2]
```diff
@@ -62,19 +62,21 @@ func (hs *HTTPServer) QueryMetricsV2(c *models.ReqContext) response.Response {
...
+		for _, res := range qdr.Responses {
+			if res.Error != nil {
+				statusCode = http.StatusBadRequest
+			}
 		}
+		return response.JSONStreaming(statusCode, qdr)
 	}
 
-	return response.JSONStreaming(statusCode, qdr)
+	res := map[string]queryResponse{}
```
Comment (ryantxu) [inline pkg/api/metrics.go:75]: why create a new one?

Comment (wbrowne) [reply pkg/api/metrics.go:75]: The idea is that the `featuremgmt.FlagDatasourceQueryMultiStatus` dictates whether or not to use the new response format IE HTTP 207 multi status header with a status field for each each response

Comment (ryantxu) [reply pkg/api/metrics.go:75]: I just mean use the [QueryDataResponse](https://github.com/grafana/grafana-plugin-sdk-go/blob/main/backend/data.go#L75) from sdk and put whatever errors you need on it.

Just seems really odd to create and maintain a new structure that can only write, but not read 🤷

File: pkg/api/metrics_response.go (hunk 1 above)
Comment (ryantxu) [reply pkg/api/metrics_response.go:15]: interesting -- seems odd to have a few parallel representations of the same thing.  In particular when the real concern from @kylebrandt etc is that the errors are not accessible to consumers anyway.  Injecting the errors directly in the json response means we both do not have a standard way to read or write it from any consumer

File: pkg/tsdb/testdatasource/scenarios.go [hunk 3]
```diff
@@ -370,6 +377,21 @@ func (s *Service) handleRandomWalkWithErrorScenario(ctx context.Context, req *ba
 	return resp, nil
 }
 
+func (s *Service) handleDataResponseErrorStatusScenario(_ context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+	resp := backend.NewQueryDataResponse()
+
+	rand.Seed(time.Now().Unix())
```
Comment (ryantxu) [inline pkg/tsdb/testdatasource/scenarios.go:383]: can we set the seed to be based on the query time range?  This makes it possible to have a stable response we can use in testing.  But when using a "to=now" it is random behavior

**Pull request closed without merging**